drivers/npmx/npmx_shell.c

```diff
@@ -3200,6 +3469,62 @@ SHELL_STATIC_SUBCMD_SET_CREATE(sub_vbusin_status,
 SHELL_STATIC_SUBCMD_SET_CREATE(sub_vbusin, SHELL_CMD(vbus, &sub_vbusin_status, "Status", NULL),
 			       SHELL_SUBCMD_SET_END);
 
+/* Creating subcommands (level 4 command) array for command "ship config time". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship_config_time,
+			       SHELL_CMD(get, NULL, "Get ship exit time", sub_ship_config_time_get),
+			       SHELL_CMD(set, NULL, "Set ship exit time", sub_ship_config_time_set),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 4 command) array for command "ship config inv_polarity". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship_config_inv_polarity,
+			       SHELL_CMD(get, NULL, "Get inverted polarity status",
+					 sub_ship_config_inv_polarity_get),
+			       SHELL_CMD(set, NULL, "Set inverted polarity status",
+					 sub_ship_config_inv_polarity_set),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 3 command) array for command "ship config". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship_config,
+			       SHELL_CMD(time, &sub_ship_config_time, "Time", NULL),
+			       SHELL_CMD(inv_polarity, &sub_ship_config_inv_polarity,
+					 "Button invert polarity", NULL),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 4 command) array for command "ship reset long_press". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship_reset_long_press,
+			       SHELL_CMD(get, NULL, "Get long press status",
+					 sub_ship_reset_long_press_get),
+			       SHELL_CMD(set, NULL, "Set long press status",
+					 sub_ship_reset_long_press_set),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 4 command) array for command "ship reset two_buttons". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship_reset_two_buttons,
+			       SHELL_CMD(get, NULL, "Get two buttons status",
+					 sub_ship_reset_two_buttons_get),
+			       SHELL_CMD(set, NULL, "Set two buttons status",
+					 sub_ship_reset_two_buttons_set),
+			       SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 3 command) array for command "ship reset". */
+SHELL_STATIC_SUBCMD_SET_CREATE(
+	sub_ship_reset, SHELL_CMD(long_press, &sub_ship_reset_long_press, "Long press", NULL),
+	SHELL_CMD(two_buttons, &sub_ship_reset_two_buttons, "Two buttons", NULL),
+	SHELL_SUBCMD_SET_END);
+
+/* Creating subcommands (level 3 command) array for command "ship mode". */
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_ship_mode,
+			       SHELL_CMD(ship, &sub_ship_mode_ship_set, "Enter ship mode", NULL),
+			       SHELL_CMD(hibernate, &sub_ship_mode_hibernate_set,
+					 "Enter hibernate mode", NULL),
```

Comment:
Function implementing a command is the last argument, not the second. Also, there probably should be another level `set`, fo example `mode set ship`, in case `get` was needed in the future.

Reply:
`cmd_` changed.

Reply:
`ship` and `hibernate` are only tasks and there will be no `getters` for them.

Reply:
But cannot mode be checked if it is ship or hibernate?

Reply:
Based on registers, no.